checkST2110.js

```diff
@@ -971,210 +1192,487 @@ const test_30_62_4 = (sdp, params) => {
+  let errors = test_10_62_1(sdp, params);
+  if (params.verbose && errors.length == 0) {
+    console.log('Test Passed: ST 2110-22 Section 7.1 Test 1 - Conforms to ST 2110-10 for video media specification');
+  }
+  return errors;
+};
+
+const mustHaves22 = ['width', 'height', 'TP']; // Defined as mandatory in ST 2110-22
+const mustHaves9134 = ['packetmode']; // Defined as mandatory in RFC 9134
+
+// ST 2110-22 Section 7.2 Test 1 - Test all required parameters are present
+const test_22_72_1 = (sdp, params) => {
+  let [mtParams, errors] = extractMTParams(sdp, { checkDups: true });
+  for (let stream of mtParams) {
+    let keys = Object.keys(stream);
+    for (let param of mustHaves22) {
+      if (keys.indexOf(param) < 0) {
+        errors.push(new Error(`Line ${stream._line}: For stream ${stream._streamNumber}, required parameter '${param}' is missing, as per ST 2110-22 Section 7.2`));
+      }
+    }
+    for (let param of mustHaves9134) {
+      if (keys.indexOf(param) < 0) {
+        errors.push(new Error(`Line ${stream._line}: For stream ${stream._streamNumber}, required parameter '${param}' is missing, as per RFC 9134 Section 7.1`));
+      }
+    }
+  }
+  if (params.verbose && errors.length == 0) {
+    console.log('Test Passed: ST 2110-22 Section 7.2 Test 1 - All RFC 9134 Section 7.1 required parameters are present');
+  }
+  return errors;
+};
+
+// ST 2110-22 Section 7.3 Test 1 - Check for mandatory bandwidth-field in correct format
+const test_22_73_1 = (sdp, params) => {
+  let errors = [];
+  let lines = splitLines(sdp);
+  let bandwidthPresent = false;
+  for (let x = 0; x < lines.length; x++) {
+    if (lines[x].startsWith('b=')) {
+      let bandwidthMatch = lines[x].match(bandwidthPattern);
+      if (bandwidthMatch == null) {
+        errors.push(new Error(`Line ${x + 1}: Bandwidth line must be of the form 'b=<bwtype>:<bandwidth>' as per ST 2110-22 Section 7.3.`));
+        continue;
+      }
+      if (bandwidthMatch[1] != 'AS') {
+        errors.push(new Error(`Line ${x + 1}: In 'b=<bwtype>:<bandwidth>' bwtype must be 'AS' as per ST 2110-22 Section 7.3.`));
+      }
+      if (Number.isInteger(+bandwidthMatch[2]) == false) {
+        errors.push(new Error(`Line ${x + 1}: In 'b=<bwtype>:<bandwidth>' bandwidth must be specified as an integer as per ST 2110-22 Section 7.3.`));
+      }
+      bandwidthPresent = true;
+    }
+  }
+  if (!bandwidthPresent) {
+    errors.push(new Error('Required bandwidth-field \'b=<bwtype>:<bandwidth>\' is missing, as per ST 2110-22 Section 7.3.'));
+  }
+  if (params.verbose && errors.length == 0) {
+    console.log('Test Passed: ST 2110-22 Section 7.3 Test 1 - Check for mandatory bandwidth-field');
+  }
+  return errors;
+};
+
+// ST 2110-22 Section 7.4 Test 1 - Check that framerate is specified by one of accepted methods
+// For 2110-07 the logic checks that
+// - If a framerate attribute is present as a session level attribute then no other framerate indication in the SDP
```

Comment:
RFC 4566 defines `framerate` as a media-level attribute. This means it cannot appear at session level.

Reply:
Thanks for this catch. I've made changes that flag an error if framerate is specified at the session level and also that one and only one framerate is specified in media streams.